doc/whats_new/v1.4.rst

```diff
@@ -101,6 +101,10 @@ Changelog
   according to :ref:`metadata routing user guide <metadata_routing>`. :pr:`27005`
   by `Adrin Jalali`_.
 
+- |API| :class:`~compose.ColumnTransformer` now replaces `"passthrough"` with a
+  corresponding :class:`~preprocessing.FunctionTransformer` in the fitted
```

Comment:
I always wanted this change, but it is backward incompatible. I am undecided on going through a deprecation cycle.

Reply:
This is a FIX to me, and we really won't be breaking many people's code by this change.

Reply:
I find the backward incompatibility problematic. I would have considered as a fix if we would have only transformers in `transformers_` and no string. But this is not the case. I can imagine some user code relying on the "passthrough" or "drop" strings. I know that this is annoying to handle those but I would rather be more confortable with a proper deprecation cycle.

I can imagine that we could a keyword `legacy_mode_attributes` that could allow replace back the `FunctionTransformer` in the `transformers_` property and raise the deprecation warnings.

Reply:
That would mean introducing the arg, do the deprecation, then introduce a deprecation for the arg that we just added to deprecate something which should never have been there in the first place. I rather not do that and annoy very few people whose code might break.

Reply:
I guess this PR is blocked by this question: Do we consider it as a bugfix without deprecation cycle or do we implement a backwards compatible deprecation? (introduce a new arg which is immediately deprecated)

@betatim @adrinjalali @glemaitre @thomasjpfan @avm19 Opinions?

I‘m fine with the bugfix version. I guess in the end it‘s less painful for both sides.

Reply:
> Is there any Python trick that would make string comparisons `FunctionTransformer() == 'passthrough'` evaluate to True, but keep repr as intended?

Yes, you can implement `__eq__`, but I really wouldn't want to pretend those two are the same.

> I‘m fine with the bugfix version. I guess in the end it‘s less painful for both sides.

+1

Reply:
@scikit-learn/core-devs we need a decision here. 

@glemaitre are you a -1 for doing this w/o deprecation?

Reply:
I see @lorentzenchr and yourself OK to consider it as a bug fix. So let's go with it.

Reply:
I just summed up the net benefit of both paths und the bugfix version won with less estimated trouble in total/for everyone.

Reply:
It is what it is, but I'm going to throw in the opinion that this should have gone through a deprecation cycle. I have a production pipeline containing a very long-running incremental estimator that can't predict on 1.4.0, and is nontrivial to retrain from scratch.